UPGRADE.md

```diff
@@ -8,9 +8,14 @@ awareness about deprecated code.
 
 # Upgrade to 3.3
 
+## Deprecated `Doctrine\DBAL\Types\Type::getName()`
+
+Instead of relying on the name as returned by this method, it is recommended to
+rely on either the class name or the name of the type inside the type registry.
```

Comment:
This is a clear violation of the Tell-Don't-Ask principle. As long consumers have to "ask" your abstraction something, you cannot evolve it w/o the risk of breaking compatibility.

Reply:
I'll reuse something similar as an upgrade note, thanks!

Reply:
> it is recommended to rely on either the class name or the name of the type inside the type registry.

It is still a violation of this principle. The type carries logic, not identity. Its identity isn't supposed to be relied on in any form.